highfidelity/hifi
# Snapshot sharing improvements

This PR modifies the snapshot behavior:
1. The Snapshot Review dialog now appears as soon as the snapshot is finished capturing (versus the current behavior, where it appears after the GIF finishes processing). While the GIF is processing, the Snapshot Review dialog displays a "Loading" image until the GIF is finished processing.

*Test Plan:*
1. From a clean install of Interface A in Desktop mode, press the "SNAP" button in the toolbar.
2. Verify that the following behavior occurs:
The Snapshot Review dialog now appears as soon as the snapshot is finished capturing.
While the GIF is processing, the Snapshot Review dialog displays a "Loading" image until the GIF is finished processing.
At that time, the captured GIF appears in the Snapshot Review dialog.
3. Check the box next to the GIF, then press "SHARE IN FEED". Wait a few seconds (up to a minute) until the GIF appears in the feed. If it never appears, click the "SNAP" button at the top of the feed, and see if it's there.
4. Open Settings -> General. UNCHECK the option next to "Take Animated GIF Snapshot", then save changes.
5. Press the "SNAP" button in the toolbar.
6. The Snapshot Review dialog appears almost immediately. There are no checkboxes in the dialog. Pressing "SHARE IN FEED" causes the still image to appear in the feed (click the "SNAP" button at the top of the feed if it doesn't automatically appear).

Verify that, throughout this test plan, HUD elements disappeared and reappeared as expected (i.e. disappeared during snapshot capture; reappeared when the review dialog appeared).
